### PR TITLE
Allow the language selector to detect autonym by exonym

### PR DESCRIPTION
**Pull request by chukarave**
This PR adds the ability to search for a language name in any language and have the autonym suggested (same behavior as Wikidata). 

Bug: T353727

### CODE REVIEW

**Review by hasanakg — COMMENTED** [review]
Everything works great! One exception: I think we need to have a debounce period for API searches, right now we send an API call immediately after key down, but we need to call the API if the user doesn't write for some period like 300ms.

File: resources/js/Components/LanguageSelector.vue [hunk 1]
```diff
@@ -73,22 +74,22 @@ const shownLanguages = computed<Language[]>(() => {
...
             params: {
-                search: inputValue,
+                search: debouncedInputValue,
                 origin: '*' // avoid CORS console errors
             }
         }).then((response) => {
             apiLanguageCodes.value = Object.keys(response.data.languagesearch);
         });
-}
+}, 200);
```
Comment (hasanakg) [inline resources/js/Components/LanguageSelector.vue:92]: For me, 200ms is good, but should we give a heads-up to the UX team?

File: resources/js/Components/LanguageSelector.vue [hunk 2]
```diff
@@ -73,22 +74,22 @@ const shownLanguages = computed<Language[]>(() => {
 
 function onInput(searchedLanguage: string): void {
 	searchInput.value = searchedLanguage;
```
Comment (hasanakg) [inline resources/js/Components/LanguageSelector.vue:76]: If I write something and then remove it, `searchInput.value` is empty and the code makes an API call with an empty language code string.

**Review by hasanakg — CHANGES_REQUESTED** [review]
We need to solve the empty input problem that I mentioned in the code comments.

File: resources/js/Components/LanguageSelector.vue (hunk 2 above)
Comment (chukarave) [reply resources/js/Components/LanguageSelector.vue:76]: Not good! should work better now

File: resources/js/Components/LanguageSelector.vue (hunk 1 above)
Comment (chukarave) [reply resources/js/Components/LanguageSelector.vue:95]: I discussed this last week with Sarai and it was decided there is not need to plan changes to the UX flow until deemed necessary in product verification. see [here](https://phabricator.wikimedia.org/T353727#9554935).

**Review by hasanakg — APPROVED** [review]

**Comment by guergana** [comment]
@chukarave as much as I would have liked to come up with the solution for the svg-jest mock transformer by myself, i actually found it in this thread, in case you wanna take a look: https://github.com/jpkleemans/vite-svg-loader/issues/35#issuecomment-1341893590

**Comment by chukarave** [comment]
@guergana nice! thanks. it's pretty straight forward, I like that it doesn't require installing any svg tansformer packages.

**Review by hasanakg — APPROVED** [review]

**Pull request merged**